### PR TITLE
Updated test-unit.yml to comment a coverage report

### PR DESCRIPTION
**Pull request by tannerjones4075**
### Description
Comment coverage report for unit tests coverage

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/test-unit.yml [hunk 1]
```diff
@@ -18,22 +18,27 @@ jobs:
...
-          file: tmp/unit.cov
-          flags: unit
-        if: matrix.os == 'ubuntu-latest'
-
+          file: coverage.out
+      - name: Comment coverage report
+        run: |
+          if [ "${{ matrix.os }}" == "ubuntu-latest" ] || [ "${{ matrix.os }}" == "windows-latest" ] || [ "${{ matrix.os }}" == "macOS-latest" ]; then
+          curl -sSfL https://codecov.io/bash | bash -s -- -Z -K -f coverage.out || echo "Codecov upload failed"
+          fi
```
Comment (dblock) [inline .github/workflows/test-unit.yml:33]: Feels like this should be the uploader as mentioned in https://about.codecov.io/blog/getting-started-with-code-coverage-for-golang/.

File: CHANGELOG.md [hunk 2]
```diff
@@ -47,6 +47,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Updated Developer guide to include docker build instructions ([#385](https://github.com/opensearch-project/opensearch-go/pull/385)
 - Test against version 2.9.0,2.10.0, run tests in all branches, change intergration tests to wait for OpenSearch to start ([#392](https://github.com/opensearch-project/opensearch-go/pull/392))
 - Makefile: use docker golangci-lint, run integration test on `.` folder, change coverage generation ([#392](https://github.com/opensearch-project/opensearch-go/pull/392)) 
+- test-unit.yml: added PR comment for unit test coverage report ([#407](https://github.com/opensearch-project/opensearch-go/pull/407))
```
Comment (dblock) [inline CHANGELOG.md:50]: This change seems like uploading results to CodeCov, no?

Comment (tannerjones4075) [reply CHANGELOG.md:50]: I am trying to understand the [output](https://app.codecov.io/gh/opensearch-project/opensearch-go/pull/407/indirect-changes#92a0d278e7f05c701b6cb6145a79652d-L432). Is it running it twice?

File: .github/workflows/test-unit.yml (hunk 1 above)
Comment (tannerjones4075) [reply .github/workflows/test-unit.yml:33]: Ok I will look into that.

**Pull request closed without merging**